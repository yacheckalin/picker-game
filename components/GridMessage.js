import React from "react";
import PropTypes from "prop-types";

const GridMessage = React.memo(({ log }) => (
  <div className="col s12 card-panel">
    <blockquote>{log}</blockquote>
  </div>
));

GridMessage.propTypes = {
  log: PropTypes.string.isRequired,
};

export default GridMessage;
