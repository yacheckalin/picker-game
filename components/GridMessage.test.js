import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import GridMessage from "./GridMessage";

describe("<GridMessage />", () => {
  const props = {
    log: "Some log message here",
  };

  test("renders and displays properly", () => {
    const wrapper = shallow(<GridMessage {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("render log message properly", () => {
    const wrapper = shallow(<GridMessage {...props} />);
    expect(wrapper.find(props.log)).toBeTruthy();
  });
});
