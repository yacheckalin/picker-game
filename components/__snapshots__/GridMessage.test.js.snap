// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridMessage /> renders and displays properly 1`] = `
<div
  className="col s12 card-panel"
>
  <blockquote>
    Some log message here
  </blockquote>
</div>
`;
